lib/pbench/agent/tool_data_sink.py

```diff
@@ -186,6 +190,408 @@ def terminate(self):
+            )
+
+    def start_logging(self):
+        """ invoke pmlogger and start logging using pmlogger_check -c """
+        # code to start logging
+        self.logger.info("starting logger")
+
+        command = PCP.pmGetConfig("PCP_BINADM_DIR") + "/pmlogger_check -c {}".format(
+            self.control_file
+        )
+        self.logger.debug("start logger command:%s", command)
+        out = subprocess.Popen(
+            [command], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True
+        )
+        stdout, stderr = out.communicate()
+        self.logger.debug("start logger command stdout:%s", stdout.decode())
+
+        if stdout.decode() != "":
+            # pmlogger failed to start. Log it
+            self.logger.error("not able to start logger successfully")
+        else:
+            self.logger.info("logger started successfully")
+
+    def stop_logging(self):
+        """ stop logging using pmlogger_check -s """
+        # code to stop logging
+        self.logger.info("stop logger")
+
+        command = PCP.pmGetConfig("PCP_BINADM_DIR") + "/pmlogger_check -s -c {}".format(
+            self.control_file
+        )
+        self.logger.debug("stop logger command:%s", command)
+        out = subprocess.Popen(
+            [command], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True
+        )
+        stdout, stderr = out.communicate()
+        self.logger.debug("stop logger command stdout:%s", stdout.decode())
+
+        if stdout.decode() != "":
+            # pmlogger failed to stop. Log it
+            self.logger.error("not able to stop logger successfully")
+        else:
+            self.logger.info("logger stopped successfully")
+
+        # All processes done, do cleanup
+        self.cleanup()
+
+    def check_connection(self, hosts):
+        """ Check if pmcd is running in given list of hosts"""
+        self.logger.info("checking connection")
+
+        for idx, host in enumerate(hosts):
+            res = ""
+            try:
+                self.logger.debug("checking connection on host:%s", host)
+                res = pmapi.pmContext(api.PM_CONTEXT_HOST, host + ":44321")
+            except Exception:
+                res = None
+
+            if res is not None:
+                self.logger.info("Port 44321 is open in:{}".format(host))
+                self.hosts.append(
+                    host
+                )  # logging only on those hosts with port 44321 active
+            else:
+                self.logger.error("Port 44321 is not open in:{}".format(host))
+                self.logger.error("Ignoring host:%s", host)
+
+    def build_pmlog_configs(self, hosts):
+        """ Build pmlog config files for each host. Returns true/false depending on success or failure in building files """
+        self.logger.info("building pmlogger config files")
+
+        for index, host in enumerate(hosts):
+            # make a folder with name self.benchmark_run_dir/temp/host
```

Comment:
According to the code below, it's `.temp`. It also looks a bit odd having the variable name inline like that. Maybe `with name {self.benchmark_run_dir}/temp/host` to set it off?